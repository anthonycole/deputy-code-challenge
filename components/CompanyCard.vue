<template>
  <article class="card">
    <div class="card-company-image-container">
      <img class="card-company-image" :src=company.image_url :alt="'Logo for ' + company.company_name" />
    </div>
    <div class="card-company-info">
      <h1 class="sr-only">{{ company.company_name }}</h1>
      <p tabindex="0" class="card-company-info-text">{{ company.excerpt }}</p>
    </div>
    <div class="card-read-more-container">
      <p class="card-read-more-link-container">
        <a href="#" class="card-read-more-link">{{ $t("read_more") }}</a>
        <img class="card-read-more-link-image" src="~/assets/images/arrow-right-blue.svg" alt="Read More Image" />
      </p>
      <p class="card-reading-time-container-link">
        <img class="card-reading-time-image" src="~/assets/images/clock.svg" alt="Clock Image " />{{ $t("min_read", {count: calculateReadingTime(company.word_count)}) }}
      </p>
    </div>
  </article>
</template>

<script>
import { AVERAGE_READING_TIME } from '../constants';

export default {
  props: ['company'],
  methods: {
    calculateReadingTime(time) {
      return Math.round(time/AVERAGE_READING_TIME);
    }
  },
}
</script>

<style scoped>
  .card {
    display: flex;
    flex-direction: column;
    align-items: center;
    background: var(--color-white);
    padding: 2.25rem;
  }

  .card-company-image-container {
    min-height: 7rem;
    max-height: 7rem;
  }

  .card-company-image {
    object-fit: contain;
    height: 5rem;
    max-height: 5rem;
  }

  .card-company-info {
    margin-top: 1rem;
    margin-bottom: 1rem;
    flex-grow: 1;
  }

  .card-company-info-text {
    color: var(--color-text);
  }

  .card-read-more-container {
    display: flex;
    flex-direction: column;
  }

  .card-read-more-link-container {
    display: inline-flex;
    margin-bottom: 0.5rem;
  }

  .card-read-more-link-image {
    margin-left: 0.5rem;
  }

  .card-read-more-link {
    color: var(--color-links);
    font-weight: bold;
  }

  .card-reading-time-container {
    margin-right: 0.5rem;
  }

  .card-reading-time-image {
    margin-right: 0.5rem;
  }

  .card-reading-time-container-link {
    color: var(--color-text);
    display: inline-flex;
    letter-spacing: 0.05em;
    text-transform: uppercase;
    font-size: 0.75rem;
    line-height: 1rem;
  }

  @media (min-width: 768px) {
    .card {
      height: 100%;
      margin-bottom: 0;
    }
  }
</style>
