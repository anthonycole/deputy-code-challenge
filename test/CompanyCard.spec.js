import { shallow } from "vue-test-utils";
import CompanyCard from "@/components/CompanyCard.vue";

describe("Card Component", () => {
  let cmp;

  beforeEach(() => {
    cmp = shallow(CompanyCard, {
      propsData: {
        company: {
          company_name: 1,
          image_url: "http://placekitten.com/200/300",
          company_size: "1000+",
          excerpt: "Lol",
        },
      },
      mocks: {
        $t: () => {},
      },
    });
  });

  it("Has the expected markup", () => {
    expect(cmp.element).toMatchSnapshot();
  });
});
