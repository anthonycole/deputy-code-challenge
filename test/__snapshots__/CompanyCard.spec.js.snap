// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Has the expected markup 1`] = `
<article
  class="mb-4 md:mb-0 md:h-full justify-between flex flex-col items-center bg-white p-10"
>
  <div
    class="min-h-28 max-h-28"
  >
    <img
      alt="1"
      class="object-contain h-20 max-h-20"
      src="http://placekitten.com/200/300"
    />
  </div>
   
  <div
    class="my-4"
  >
    <h1
      class="sr-only"
    >
      1
    </h1>
     
    <p
      class="text-brand-text"
      tabindex="0"
    >
      Lol
    </p>
  </div>
   
  <div
    class="flex flex-col"
  >
    <p
      class="inline-flex mb-4"
    >
      <a
        class="text-brand-links font-bold"
        href="#"
      >
        Read More
      </a>
       
      <img
        alt="Read More"
        class="ml-2"
        src="~/assets/images/arrow-right-blue.svg"
      />
    </p>
     
    <p
      class="inline-flex text-xs text-brand-text uppercase tracking-wider"
    >
      <img
        alt="Read More"
        class="mr-2"
        src="~/assets/images/clock.svg"
      />
      NaN min read
    
    </p>
  </div>
</article>
`;
